### ISSUE TITLE
No macro.c

### ISSUE THREAD

**Issue by lkardos** [lkardos/test#94]
I recently tried to emerge rpm in gnetoo linux with a 2.6.6-r6 kernel and the emerge fails.  It shows an error with macro.c (listed below).  It was seuggested to look at the code for macro.c and see where the issue is.  Using the find command nothing turned uo for macro.c.  looks like the file is not present in the package.    macro.c: In function rpmGlob:   macro.c:1700: warning: implicit declaration of function setlocale   macro.c:1700: error: LC_COLLATE undeclared (first use in this function)   macro.c:1700: error: (Each undeclared identifier is reported only once   macro.c:1700: error: for each function it appears in.)   macro.c:1700: warning: assignment makes pointer from integer without a cast   macro.c:1703: error: LC_CTYPE undeclared (first use in this function)   macro.c:1703: warning: assignment makes pointer from integer without a cast

**Comment by lkardos** [comment]
_Imported from trac issue 94.  Created by scottneh on 2009-09-08T20:53:20, last modified: 2014-01-24T11:59:53_

**Comment by lkardos** [comment]
_Trac comment by pmatilai on 2009-09-15 10:58:54:_

Err.. non-existent file would not complain about having undeclared symbols in it.    The actual issue is that ./configure is not finding locale.h header for whatever reason_ have a look at config.log for details. Once you fix that it should build just fine.    The other issue (a real bug_ but a rather minor one) is that rpm doesn't handle missing locale.h cleanly despite sorta trying to.

**Comment by lkardos** [comment]
_Trac comment by pmatilai on 2014-01-24 11:59:53:_

WONTFIX in the sense that 4.4.x is long_ long since dead_ was even in 2009. --disable-nls allows working around a missing locale.h at least in newer versions_ but I'd think locale.h missing suggests some other problems on the system as its a standard header.
